modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2819,6 +2824,13 @@ protected void updateModelForComposedSchema(CodegenModel m, Schema schema, Map<S
         // end of code block for composed schema
     }
 
+    @SuppressWarnings("unchecked")
+    private static String getEnumStringLiteralUnionType(Schema interfaceSchema) {
+        return (String) interfaceSchema.getEnum().stream()
+                .map(enumName -> "\"" + (String)enumName + "\"")
+                .collect(Collectors.joining(" | "));
```

Comment:
is that the same as https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L1123 ?

cc @OpenAPITools/generator-core-team